deployments/systemd/packages/rpm/SPECS/nvidia-mig-manager.spec

```diff
@@ -132,8 +132,11 @@ function maybe_remove_config_symlink() {
   fi
 }
 
-maybe_remove_hooks_symlink
-maybe_remove_config_symlink
+if [ $1 -eq 0 ]
```

Comment:
Admittedly, I am not super familiar with RPM spec files, but the documentation suggests that `%postun` is executed during package removal, and it does not mention anything about upgrades:
```
- postun

Denotes the dependency must be present right after the package is is removed, and is used a strong ordering hint to break possible dependency loops.

Also relates to %postun scriptlet execution.
```
https://rpm-software-management.github.io/rpm/manual/spec.html

Is this also executed during package upgrades? To me, it would make sense to unconditionally remove the symlinks (if they exist) in `%postun` since just a few lines earlier we are disabling the mig manager service. During an upgrade, the symlinks could be re-added in a different step.

Reply:
let me look into this a bit more

Reply:
@cdesiniotis This was  the log from an update. 

```
D: %post(nvidia-mig-manager-0.10.0-1.x86_64): waitpid(3057525) rc 3057525 status 0
ufdio:       5 reads,     8468 total bytes in 0.000006 secs
D: ========== +++ nvidia-mig-manager-0.7.0-1 x86_64-linux 0x2
D:     erase: nvidia-mig-manager-0.7.0-1.x86_64 has 15 files
D: Plugin: calling hook psm_pre in selinux plugin
D: %preun(nvidia-mig-manager-0.7.0-1.x86_64): scriptlet start
fdio:       2 writes,      726 total bytes in 0.000010 secs
D: %preun(nvidia-mig-manager-0.7.0-1.x86_64): execv(/bin/sh) pid 3057570
D: Plugin: calling hook scriptlet_fork_post in selinux plugin
D: setexecfilecon: (/bin/sh, rpm_script_t)
+ systemctl disable nvidia-mig-manager.service
Removed "/etc/systemd/system/multi-user.target.wants/nvidia-mig-manager.service".
Removed "/etc/systemd/system/nvidia-gpu-reset.target.wants/nvidia-mig-manager.service".
+ systemctl daemon-reload
+ maybe_remove_hooks_symlink
++ readlink -f /etc/nvidia-mig-manager/hooks.yaml
+ local target=/etc/nvidia-mig-manager/hooks-minimal.yaml
+ '[' /etc/nvidia-mig-manager/hooks-minimal.yaml = /etc/nvidia-mig-manager/hooks-minimal.yaml ']'
+ rm -rf /etc/nvidia-mig-manager/hooks.yaml
+ '[' /etc/nvidia-mig-manager/hooks-minimal.yaml = /etc/nvidia-mig-manager/hooks-default.yaml ']'
+ maybe_remove_config_symlink
++ readlink -f /etc/nvidia-mig-manager/config.yaml
+ local target=/etc/nvidia-mig-manager/config-default.yaml
+ '[' /etc/nvidia-mig-manager/config-default.yaml = /etc/nvidia-mig-manager/config-default.yaml ']'
+ rm -rf /etc/nvidia-mig-manager/config.yaml
```
so you see it does the post of the newer package before the preun of the old package.  I am not sure the upgrade scenario works if you are removing the links in preun/postun as that will come after the new install. 

What is the concern with the solution as is?  Did you see these references?

https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#_syntax
https://fedoraproject.org/wiki/PackagingDrafts/ScriptletSnippets#Initscripts_in_spec_file_scriptlets

Reply:
Thanks for the references, that clarifies a lot.

One question -- is the mig-manager service restarted during the upgrade? I see we disable the mig-manager service during `preun` but do we ever re-enable it after?

Reply:
Based on the scriplet ordering https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#ordering it does not appear we run any scriplet after `prerun`. Is my understanding correct?

Reply:
Got it. But by disabling the `nvidia-mig-manager.service` in `preun`, doesn't that mean the `nvidia-mig-manager.service` would not run at all if the node was rebooted? 

Maybe we need to wrap lines 115-116 in the same conditional so that they only are executed on an uninstall of the mig-parted package, and not on an upgrade?

Reply:
What I am getting at is that `prerun` should be a no-op for package upgrades. We should not remove the `config.yaml` / `hooks.yaml` symlinks AND we should not be disabling the `nvidia-mig-manager.service`.

Reply:
this is fixed now